engine/alternative-runtimes.md

```diff
@@ -0,0 +1,285 @@
+By default, containerd uses runc as its container runtime.
+
+containerd supports a
+[shim API](https://github.com/containerd/containerd/blob/main/runtime/v2/README.md){: target="blank" rel="noopener"}
+which lets you select a different container runtime.
+
+Examples of alternative container runtimes that you can use include:
+
+- [youki](https://github.com/containers/youki){: target="blank" rel="noopener"}
+- [Wasmtime](https://wasmtime.dev/){: target="blank" rel="noopener"}
+- [gVisor](https://github.com/google/gvisor){: target="blank" rel="noopener"}
+- [Kata Containers](https://github.com/kata-containers/kata-containers){: target="blank" rel="noopener"}
+
+## How to use alternative runtimes
+
+There are different ways you can go about selecting an alternative container
+runtime.
+
+- [Configure the default runtime](#configure-the-default-runtime)
+- [Register additional runtimes](#register-additional-runtimes)
+- [Use containerd shims](#use-containerd-shims)
+
+### Configure the default runtime
+
+You can change the default runtime of the daemon, using the `default-runtime`
+option.
+
+The following examples show how to configure the daemon to use gVisor (`runsc`)
+as the default runtime:
+
+- Using the daemon configuration file (`/etc/docker/daemon.json`)
+
+  ```json
+  {
+    "default-runtime": "/usr/bin/runsc"
+  }
+  ```
+
+- Using the `dockerd` CLI
+
+  ```console
+  $ dockerd --default-runtime /usr/bin/runsc
+  ```
+
+### Register additional runtimes
+
+You can register additional runtimes with the Docker daemon on startup.
+This doesn't change the default runtime, but makes it possible to select
+alternative runtimes on a per-container basis.
+
+The following examples show how to register `youki` as a container runtime:
+
+- Using the daemon configuration file (`/etc/docker/daemon.json`)
+
+  ```json
+  {
+    "runtimes": {
+      "youki": {
+        "path": "/usr/bin/youki",
+        "runtime-args": ["--debug"]
+      }
+    }
+  }
+  ```
+
+- Using the `dockerd` CLI
+
+  ```console
+  $ dockerd --add-runtime youki=/usr/bin/youki
+  ```
+
+> **Note**
+>
+> You can specify additional arguments when you register a runtime using the
+> configuration file, but not when you use the CLI flag.
+
+### Use containerd shims
+
+Docker Engine supports the use of containerd runtime shims.
+Using shims, you can use alternative runtimes without having to configure or
+restart the Docker daemon.
+
+To use a containerd shim, you must install the shim binary on `PATH` on the
+system where the Docker daemon is running. With the shim installed,
+you can select the shim with `docker run` using the shim's fully qualified name.
+Fully qualified shim names take the form of `io.containerd.<shim>.<version>`.
+
+The following `docker run` command uses the Kata Containers runtime, provided
+that the shim binary is installed on `PATH`.
+
+```console
+$ docker run --runtime io.containerd.kata.v2 hello-world
+```
+
+#### Configure shims
```

Comment:
When `runtimeType` is `io.containerd.runc.v1` or `io.containerd.runc.v2`, [these are the available options.](https://github.com/containerd/containerd/blob/326cd0623eb99aa6c0fd6b1255add105377d789e/docs/cri/config.md?plain=1#L350-L382) Though most users of the runC shim would be best served by [letting the daemon configure the runC shim options automatically](https://github.com/moby/moby/blob/32acecbf5ee3dd924313ae933d40ceac7654163a/daemon/runtime_unix.go#L58-L67) by configuring a runtime using `"path"` instead of `"runtimeType"` so it's probably better not to mention it here, and only document those options in reference documentation for advanced users, or not document it at all.